IllegalStateException is thrown when Liberty tries to update a readOnly subject

**Describe the bug**

The `java.lang.IllegalStateException` is thrown with the reason `Subject is read-only`.  The error is thrown when TAI authenticator tries to check and remove an internal property that is no longer used.   

```
Caused by: java.lang.IllegalStateException: Subject is read-only
        at javax.security.auth.Subject$SecureSet$1.remove(Subject.java:1297)
        at javax.security.auth.Subject$SecureSet.remove(Subject.java:1386)
        at java.util.Collections$SynchronizedCollection.remove(Collections.java:2051)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.removeInternalProps(TAIAuthenticator.java:388)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.authenticateWithSubject(TAIAuthenticator.java:374)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.loginWithTAIUserName(TAIAuthenticator.java:351)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.authenticateWithTAIResult(TAIAuthenticator.java:313)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.handleTaiResult(TAIAuthenticator.java:252)
        at com.ibm.ws.webcontainer.security.internal.TAIAuthenticator.authenticate(TAIAuthenticator.java:167)
        at com.ibm.ws.webcontainer.security.WebProviderAuthenticatorProxy.handleTAI(WebProviderAuthenticatorProxy.java:337)
        at com.ibm.ws.webcontainer.security.WebProviderAuthenticatorProxy.authenticate(WebProviderAuthenticatorProxy.java:454)
        at com.ibm.ws.webcontainer.security.WebAuthenticatorProxy.authenticate(WebAuthenticatorProxy.java:68)
        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.authenticateRequest(WebAppSecurityCollaboratorImpl.java:1233)
        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.determineWebReply(WebAppSecurityCollaboratorImpl.java:989)
        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.performSecurityChecks(WebAppSecurityCollaboratorImpl.java:678)
        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.preInvoke(WebAppSecurityCollaboratorImpl.java:596)
        at com.ibm.wsspi.webcontainer.collaborator.CollaboratorHelper.preInvokeCollaborators(CollaboratorHelper.java:459)

```

**Steps to Reproduce**
Unfortunately, no recreation steps are found.  

**Expected behavior**
Liberty should check isReadOnly() before accessing the Subject.  

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.12 
 - Java Version: The issue is irrelevant to the Java version.  Reported on 8.0.6.25. 
 - server.xml configuration  The error is thrown from TAI related method.  The TAI configuration looks like below. 
 ```
  <trustAssociation id="myTrustAssociation" invokeForUnprotectedURI="false" failOverToAppAuthType="false">
    <interceptors id="myTAI" enabled="true" className="com.ibm.mysecurity.myTAI" invokeBeforeSSO="true" invokeAfterSSO="false" libraryRef="myLib">
    </interceptors>
  </trustAssociation>
```

--- a/dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/TAIAuthenticator.java
+++ b/dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/internal/TAIAuthenticator.java
@@ -381,7 +381,7 @@ public class TAIAuthenticator implements WebAuthenticator {
 
     protected void removeInternalProps(Subject subject, SubjectHelper subjectHelper, String propName) {
         Hashtable<String, Object> hashtable = (Hashtable<String, Object>) subjectHelper.getSensitiveHashtableFromSubject(subject);
-        if (hashtable == null)
+        if (hashtable == null || subject.isReadOnly())
             return;
 
         Set<Object> publicCredentials = subject.getPublicCredentials();

